test/Enum.rb

```diff
@@ -92,6 +92,14 @@ def test_issue593_classtype
     end
   end
 
+  def test_issue593_filter_types
```

Comment:
I think I'd probably name these tests something more descriptive, like `test_using_filter_types_does_not_cause_endless_loop`, and then maybe link back to the original issue in a comment. Though really, a user could also track back to the original issue via the commit without needing a link.

Reply:
I think you are right. Will do that differently in other PR's that are coming and renamed the current ones in #620